src/ConfigWidgets/SymbolsDialog.cpp

```diff
@@ -88,4 +109,85 @@ void SymbolsDialog::OnRemoveButtonClicked() {
   }
 }
 
+std::tuple<QString, QString> SymbolsDialog::GetFilePickerConfig() const {
+  QString file_filter{"Symbol Files (*.debug *.so *.pdb *.dll);;All files (*)"};
+
+  if (!module_.has_value()) {
+    return std::make_tuple("Select symbol file", file_filter);
+  }
+
+  const ModuleData& module{*module_.value()};
+
+  QString caption =
+      QString("Select symbol file for module %1").arg(QString::fromStdString(module.name()));
+
+  switch (module.object_file_type()) {
+    case ModuleInfo::kElfFile:
+      file_filter = "Symbol Files (*.debug *.so);;All files (*)";
+      break;
+    case ModuleInfo::kCoffFile:
+      file_filter = "Symbol Files (*.pdb, *.dll);;All files (*)";
+      break;
+    default:
+      ORBIT_ERROR("Cant determine file picker filter: unknown module type");
+      break;
+  }
+
+  return std::make_tuple(caption, file_filter);
+}
+
+void SymbolsDialog::OnAddFileButtonClicked() {
+  QSettings settings;
+
+  auto [caption, file_filter] = GetFilePickerConfig();
+
+  QString file = QFileDialog::getOpenFileName(
+      this, caption, settings.value(kFileDialogSavedDirectoryKey).toString(), file_filter);
+  if (file.isEmpty()) return;
+
+  std::filesystem::path path{file.toStdString()};
+
+  settings.setValue(kFileDialogSavedDirectoryKey,
+                    QString::fromStdString(path.parent_path().string()));
+  ErrorMessageOr<void> add_result = TryAddSymbolFile(path);
+
+  if (!add_result.has_error()) return;
+
+  QMessageBox::warning(this, "Unable to add file",
+                       QString::fromStdString(add_result.error().message()));
+}
+
+ErrorMessageOr<void> SymbolsDialog::TryAddSymbolFile(const std::filesystem::path& file_path) {
+  ErrorMessageOr<std::unique_ptr<orbit_object_utils::SymbolsFile>> symbols_file_or_error =
+      orbit_object_utils::CreateSymbolsFile(file_path, orbit_object_utils::ObjectFileInfo());
+
+  if (symbols_file_or_error.has_error()) {
+    return ErrorMessage(absl::StrFormat("The selected file is not a viable symbol file, error: %s",
+                                        symbols_file_or_error.error().message()));
+  }
+
+  const orbit_object_utils::SymbolsFile& symbols_file{*symbols_file_or_error.value()};
+
+  const std::string build_id = symbols_file.GetBuildId();
+  if (build_id.empty()) {
+    return ErrorMessage("The selected file does not contain a build id");
+  }
+
+  if (!module_.has_value()) return TryAddSymbolPath(file_path);
```

Comment:
This deserves a comment. Do I get this right? If we aren't adding the symbols for a specific module then we just add the whole directory?

Reply:
No. (I also added a comment in the code). 

This check is about whether the dialog was opened with a module or not. If it was opened with a module (Also contains the label: Add symbols for [module]), then the build-ids are compared, to check whether the symbol file is suited for the module. If the dialog is opened without a module, this build id matching does not take place.